### ISSUE TITLE
MineOS не запускается после обновления

### ISSUE THREAD

**Issue by Senliast** [IgorTimofeev/MineOS#320]
Не всегда но часто после обновления MineOS не запускается, при запуске появляется ошибка. После этого надо использовать восстановление по интернету чтобы ОС снова запустилась. Проявлялось уже много раз на нескольких компах. Почему?

Ошибка: https://imgur.com/a/DXY92I6

**Comment by IgorTimofeev** [comment]
Судя по всему, подгружается какая-то левая библиотека Filesystem, заменяя оригинальную. В обновленной отсутствует функция setProxy. Никакой левый софт с маркета не стоит?

**Comment by Senliast** [comment]
Нет. Как я сказал, встречал на нескольких компах, ничего стороннего не установлено.

**Comment by IgorTimofeev** [comment]
Ну пиздец. Хуй пофиксишь тогда без деталей, сорре

**Comment by Senliast** [comment]
А я могу чем помочь? Могу логи какие прислать или что... Если это поможет - у меня MC 1.7.10, последняя версия OpenComputers. Может быть у меня версия майна старая и для нее специально надо оптимизировать?

**Comment by MrFredber** [comment]
Дополнение: именно в майне

**Comment by IgorTimofeev** [comment]
@Senliast, да черт его знает, правда. Помочь вряд ли тут чем-то можно, так как проблема хаотична и рандомно воспроизводима. Можно разве что прислать "сломанный" файл Libraries/Filesystem.lua, чтобы узнать, почему в нем отсутствует функция setProxy или хотя бы вычислить его происхождение. 

Я вообще склоняюсь к версии, что в маркете возник какой-то конфликт IDшников, но чтобы его разрешить, нужна какая-то отправная точка - хотя бы в виде "сломанного" файла

**Comment by Senliast** [comment]
В следующий раз когда это будет - я вытащу файл и прикреплю его здесь. У меня кажется есть ещё компы со старыми версиями ОС, вот я их обновлю и посмотрю.

**Comment by IgorTimofeev** [comment]
Хорошо, спасибо

**Comment by Senliast** [comment]
Вот полное содержимое диска до и после обновления. Хочу заметить, что в списке обновления нет файла "Filesystem.lua". А другой комп, для которого обновление этого файла было, обновился и потом запустился нормально. Я посмотрел в этот файл после обновления и в нем содержимое из "Event.lua". Поэтому я думаю, ты прав с конфликтом ID или я не знаю но по какой-то причине содержимое из Event оказывается в Filesystem.

https://mega.nz/file/lMVHzC6L#QHtiJmibeLr3iusCAI1uk8tNeEpu7wxlcybZ6rdFyE4

**Comment by IgorTimofeev** [comment]
@Senliast Благодарю за подробную инфу. Как и предполагалось, качается библиотека Event вместо Filesystem:

![image](https://user-images.githubusercontent.com/11760002/107124495-53d88480-68b5-11eb-91e0-a25f1efcf9e3.png)

Проверил БД - все ID на месте, никакой магии не происходило. Проверил конфиги - тоже. Хуй знает, как фиксить, если проблема хаотична. Может быть, дохнет SD-карта на хост-машинке и периодически "скачет" с цифры на цифру при чтении (ID файла Event в БД 1063, ID Filesystem - 1064. Оба поля индексированные, мб индексы шалят). Иссуя пусть будет открыта, но я рили хз, как это воспроизводить, сорре

**Comment by IgorTimofeev** [comment]
Обновил IDшники в БД для Filesystem/Event, мб поможет

**Comment by Senliast** [comment]
Попробовал обновить ещё раз, теперь при загрузке Filesystem.lua и Event.lua появляется ошибка https://imgur.com/a/59Yzptv

**Comment by IgorTimofeev** [comment]
О, ну так это уже другое дело. Либо на клиенте, либо на серваке (смотря где играешь) какая-то хуйня при записи файлов на диск, майноська тут ни при чем. Конкретно этот файл не может открыться для записи по указанной на скрине причине.

А прикинь, если хост блочит открытие всех файлов с названием "Filesystem", во рофл будет

**Comment by Senliast** [comment]
Всмысле, а что тогда мешает ему открыться? У меня все останльные файлы и программы в MineOS и на реальной ОС то есть на самом компе нормально открываются, никаких битых файлов у меня никогда не было... Поэтому я не думаю что это что-то с моим жёстким диском если ты это имеешь ввиду. Но тогда почему один и тот же файл уже на нескольких комах не открывается, это же не может быть совпадением? И как хост может это блокировать это же типа в майне происходит?

**Comment by IgorTimofeev** [comment]
Блин, да откуда я-то знаю? Corrupted virtual filesystem при попытке открыть файл для записи. Ошибка выдается на уровне мода и компонента filesystem, а не ОС. Причин может быть жопой жуй: мб какая-то хуйня с правами на файлы, мб файл чем-то юзается, мб какой-нибудь лексический фильтр включён
